core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -1229,6 +1226,16 @@ private void updateSitesView(Collection<String> sitesUp, Collection<String> site
       } finally {
          viewUpdateLock.unlock();
       }
+      if (sitesUp.isEmpty()) {
+         return;
+      }
+      if (isCoordinator()) {
+         globalXSiteAdminOperations.onSitesUp(sitesUp);
+      } else {
+         //for the case where the coordinator isn't the site master.
+         //TODO improve this by checking if the coordinator is a site master or not.
```

Comment:
Is this meant to be fixed in this PR? If not, let's create a Jira and reference the id here.

Reply:
It was just an idea I had today. Not sure if JGroups exposes the info required or not.